oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapAggregator.java

```diff
@@ -173,9 +174,18 @@ private MapAggregator(MapAggregator<U, ?> obj, MapReducer<IndexValuePair<U, X>>
       SerializableFunction<X, OSHDBTimestamp> indexer
   ) {
     final TreeSet<OSHDBTimestamp> timestamps = new TreeSet<>(this.mapReducer.tstamps.get());
+    final OSHDBTimestamp minTime = timestamps.first();
+    final OSHDBTimestamp maxTime = timestamps.last();
     return this.aggregateBy(data -> {
       // match timestamps to the given timestamp list
-      return timestamps.floor(indexer.apply(data));
+      OSHDBTimestamp aggregationTimestamp = indexer.apply(data);
+      if (aggregationTimestamp.compareTo(minTime) < 0
+          || aggregationTimestamp.compareTo(maxTime) > 0) {
```

Comment:
`indexer.apply(data)` could potential return a null value.  `.aggregateByTimestamp(x -> null) `.
so maybe also at a check for that case to
```java
if ( aggregationTimestamp == null 
     || aggregationTimestamp.compareTo(minTime) < 0
     || aggregationTimestamp.compareTo(maxTime) > 0) {
```